### PR TITLE
new try with new line

### PR DESCRIPTION
**Pull request by balcy**

### CODE REVIEW

**Comment by Flohack74** [comment]
You can now go to https://ci.ubports.com/blue/organizations/jenkins/docviewer-app/detail/PR-27/6/artifacts and download and install the debs you need, or you can use our shiny new ubports-qa script to overlay them automatically. @NeoTheThird can you maybe give him the cmdline for that, I am on the run and I dont want to block testing this too long ;)

**Comment by balcy** [comment]
`sudo ubports-qa install docviewer-app 27` gave "Issue failed to build", I think it only finds build 5 but not 6 in https://ci.ubports.com/blue/rest/organizations/jenkins/pipelines/docviewer-app/branches/xenial/

**Comment by balcy** [comment]
ok I tested it manually on BQ Aquaris 4.5 RC (2018-W30)
- had to additionally install qtdeclarative5-quicklayouts-plugin 
- then since it requires libreoffice | libreoffice-vanilla I installed libreoffice and all the armhf debs of the build
- (because the root partition is quite small, I uninstalled most language packs first)
- after all debs had been installed, "Document Viewer" (with default Ubuntu app icon) appeared in the Apps scope
```
phablet@bq:/media/phablet/7873-B162/docview$ tail -f /home/phablet/.cache/upstart/application-legacy-com.ubuntu.docviewer-1533233190969400.log 
Loading module: 'libubuntu_application_api_touch_mirclient.so.3.0.0'
qml: Path of the document: /media/phablet/7873-B162/Documents/Imported/CathBaz.pdf
[FILE] Extracting information from the file...
[FILE] Requested file mime type: QVariant(QString, "application/pdf")
Loading document...
Document loaded successfully !
Populating model...
Ideal number of image providers is: 4
Loading image provider(s)...
Image provider(s) loaded successfully !
file:///usr/share/ubuntu-docviewer-app/qml/pdfView/PdfView.qml:43: TypeError: Cannot read property 'status' of null
[FILE] Path parsed as: "/media/phablet/7873-B162/Documents/Imported/CathBaz.pdf"
Number of pages: 40
Model has been successfully populated!
qml: Path of the document: /media/phablet/7873-B162/Documents/a.txt
[FILE] Extracting information from the file...
[FILE] Requested file mime type: QVariant(QString, "text/plain")
[FILE] Path parsed as: "/media/phablet/7873-B162/Documents/a.txt"
qml: Path of the document: /media/phablet/7873-B162/Documents/example.odt
[FILE] Extracting information from the file...
[FILE] Requested file mime type: QVariant(QString, "application/vnd.oasis.opendocument.text")
[FILE] Path parsed as: "/media/phablet/7873-B162/Documents/example.odt"
Loading document...
LibreOffice binaries found at: "/usr/lib/libreoffice/program"
LibreOffice profile path: "file:///home/phablet/.config/com.ubuntu.docviewer/libreoffice/4"
Document loaded successfully !
-> then app crashes
```
-> pdf and text file worked, odt showed loading and crashed

**Comment by balcy** [comment]
only did a short test: the crash was solved by using the latest LibreOfficeKit headers (ubuntu xenial repro), after adding the dependencies a sample .odt and .odg opened on the phone.
I noticed that there are some graphical artifacts on the left / right / top of the odt document, and that the color of a sample .odg document was not correct. (e.g yellow ellipse instead of blue)
Also it might be that not all dependencies are required...
I can provide a click package on Wednesday. How can I send it to in order to upload to open store ? https://open-store.io/app/com.ubuntu.docviewer It is managed by "UBPorts apps team"

**Comment by balcy** [comment]
@nfsprodriver  I have also added a clickable.json

**Comment by nfsprodriver** [comment]
Great work, thank you! Will test it later...

**Comment by Flohack74** [comment]
@balcy do you want to be the new maintainer also? See #1 :) - Then I can arrange with Brian to give you upload in  Openstore

**Comment by Flohack74** [comment]
And please send a click build into the QA Telegram group, then they can run a wider test!

**Comment by Flohack74** [comment]
I still get failed to open library '/opt/click.ubuntu.com/.click/users/phablet/com.ubuntu.docviewer/lib/arm-linux-gnueabihf/libreoffice/program/libmergedlo.so': libGLEW.so.1.13: cannot open shared object file:
See http://paste.ubuntu.com/p/GWkgpFw8BF/ - its @nfsprodriver build :)

**Comment by balcy** [comment]
Hi thank you for the feedback, I am quite busy (and on the road) today and tomorrow, and will look into it on Wednesday. Being Maintainer / Uploader to open store is ok for me too.

**Comment by nfsprodriver** [comment]
Sorry, the curl part doesn't worl inside my clickable container.

**Comment by nfsprodriver** [comment]
Sorry, just missed telegram conversation. So we should fix further things in OTA5. About version numbering we should talk about (already bumped it up from 2.1.x to 2.2.x). Latest changes from @balcy lead me to be unable to build it via clickable so I need to revert some things for building it (not affecting the click package itself). @Flohack74 or @mariogrip what do you think?

File: src/plugin/file-qml-plugin/docviewerutils.cpp [hunk 1]
```diff
@@ -76,6 +76,8 @@ QString DocviewerUtils::getXdgDocumentsLocation()
 
 QString DocviewerUtils::buildDestinationPath(const QString &destinationDir, const QString &sourcePath)
 {
+    Q_UNUSED( destinationDir )
```
Comment (mardy) [inline src/plugin/file-qml-plugin/docviewerutils.cpp]: Please keep the coding style consistent: `Q_UNUSED(destinationDir);`
(also note the semicolon at the end)

File: src/plugin/libreofficetoolkit-qml-plugin/LibreOfficeKit/LibreOfficeKitInit.h [hunk 2]
```diff
@@ -203,23 +209,43 @@ static LibreOfficeKit *lok_init_2( const char *install_path,  const char *user_p
     imp_lib = strdup("the app executable");
     dlhandle = RTLD_MAIN_ONLY;
 #endif
+    *_imp_lib = imp_lib;
+    return dlhandle;
+}
+
+typedef LibreOfficeKit *(LokHookFunction)( const char *install_path);
```
Comment (mardy) [inline src/plugin/libreofficetoolkit-qml-plugin/LibreOfficeKit/LibreOfficeKitInit.h]: Here (and in the lines below) too, please remove spaces after `(` and before `)`

**Comment by nfsprodriver** [comment]
Still unable to run the curl part in clickable. Also can't install python3-pycurl module due to dependency errors.

**Comment by nfsprodriver** [comment]
Does this PR work for anyone as it is? Still fighting up with clickable and get-click-deps.

**Comment by Flohack74** [comment]
I have no problem if this PR works only for @balcy that we merge it and bump version then. But could you send us a click build, so we can compare if it is the same quality like @nfsprodriver has?

**Comment by balcy** [comment]
@nfsprodriver  this is really strange, because curl is in the deps, does the following work for you:
```
chris@Quark:~/git/docviewer-app$ clickable run 'which curl'
Using docker container "clickable/ubuntu-sdk:16.04-armhf"
Checking dependencies
Dependencies already setup
/usr/bin/curl
```
have you tried clickable --update-docker already ?

**Comment by Flohack74** [comment]
Now I remember that for me curl also did not work despite it should be installed... I gave up on this one ;)

**Comment by balcy** [comment]
maybe deleting .clickable folder works ? I thought the docker container should behave the same way for all users ; [

**Comment by nfsprodriver** [comment]
@balcy neither clickable run 'which curl' nor clickable --update-docker commands are working for me. They don't exist. Anyway as you commited we should keep the prebuilt command for now and handle the tmp stuff etc. later. May Brian adds some helpful features for clickable soon :)

**Comment by balcy** [comment]
sorry it was `clickable update-docker`
but `clickable run 'which curl'` should work
my clickable version (`clickable --version`) is
clickable 4.4.0
it might also be helpful to delete the .clickable folder (if the version is already the latest)
yes prebuild does not hurt : )  just strange that it works for me inside docker but not for flohack and you...

**Comment by nfsprodriver** [comment]
Okay, did your steps installing latest clickable via pip, but still no changes so we should leave prebuild for now...

**Comment by balcy** [comment]
Yes I think so, I can also ask Brian if it is possible to pass the arch (armf / amd64) to the prebuild script e.g. as variable or have it as environment variable. (e.g. we might have armhf and arm64 at some point).
I think it is ready to be merged.

**Comment by Flohack74** [comment]
We are ready to merge, if you tell me I should. Then plz make me a click I can manually send to the Openstore.

**Comment by balcy** [comment]
ok I have uploaded the click package to the QA group just now.

**Pull request merged**